### PR TITLE
[arrow] Update to 16.0.0

### PR DESCRIPTION
**Pull request by raulcd**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/arrow/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "arrow",
-  "version": "15.0.2",
+  "version": "16.0.0",
   "port-version": 1,
```
Comment (Cheney-W) [inline ports/arrow/vcpkg.json:4]:
```suggestion
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Comment by Cheney-W** [comment]
```
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:34 (message):
  Applying patch failed: Checking patch cpp/CMakeLists.txt...
  error: while searching for:
    list(APPEND ARROW_SYSTEM_LINK_LIBS "ws2_32.dll")
  endif()
  if(NOT WIN32 AND NOT APPLE)
    # Pass -lrt on Linux only
    list(APPEND ARROW_SYSTEM_LINK_LIBS rt)
  endif()
  error: patch failed: cpp/CMakeLists.txt:973
  error: cpp/CMakeLists.txt: patch does not apply
  Checking patch cpp/src/arrow/vendored/musl/strptime.c...
```

**Comment by dg0yt** [comment]
> ~~~
> list(APPEND ARROW_SYSTEM_LINK_LIBS "ws2_32.dll")

I hope they changed `ws2_32.dll` to `ws2_32`. This would explain and fix https://github.com/microsoft/vcpkg/issues/38305.

File: ports/arrow/android.patch [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-index 05c05c6..f2754a5 100644
---- a/cpp/CMakeLists.txt
-+++ b/cpp/CMakeLists.txt
-@@ -973,7 +973,7 @@ if(WIN32)
+diff --git a/cpp/src/arrow/CMakeLists.txt b/cpp/src/arrow/CMakeLists.txt
+index 026bb5c..5c1b5e3 100644
+--- a/cpp/src/arrow/CMakeLists.txt
++++ b/cpp/src/arrow/CMakeLists.txt
+@@ -166,7 +166,7 @@ if(WIN32)
    list(APPEND ARROW_SYSTEM_LINK_LIBS "ws2_32.dll")
```
Comment (dg0yt) [inline ports/arrow/android.patch:6]: So upstream still has `.dll` here.  https://github.com/microsoft/vcpkg/issues/38305 remains an issue.

Comment (raulcd) [reply ports/arrow/android.patch:6]: I can open an issue upstream, @kou FYI

Comment (kou) [reply ports/arrow/android.patch:6]: Ah, you're right. We should not use `.dll` suffix. I'll fix it in upstream: #41340

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**